### PR TITLE
CodeEditor: refactors story from knobs to controls

### LINKED ISSUE #34266: Monaco: Code editor crashes in Storybook
**What happened**:
I tried running the monaco story on browser but ended up getting this error:
```
Error
[object Event]
Call Stack
 rejectionHandler
  vendors~main.iframe.bundle.js:271070:21
```
**What you expected to happen**:
I expect the component to run without any failures
**How to reproduce it (as minimally and precisely as possible)**:
Run storybook on local
Search for the codeEditor story
Click it and see how it behaves.

### PR DESCRIPTION
**Pull request by thisisobate**
**What this PR does / why we need it**:
This PR  refactors story from knobs to controls. It also fixes storybook crashing when monaco loads.
**Which issue(s) this PR fixes**:

Relates to #34417 and fixes #34266

### CODE REVIEW

File: packages/grafana-ui/src/components/Monaco/CodeEditor.story.internal.tsx [hunk 1]
```diff
@@ -38,26 +13,41 @@ export default {
...
         action('code blur')(text);
       }}
       onSave={(text: string) => {
         action('code saved')(text);
       }}
-      showLineNumbers={showLineNumbers}
-      showMiniMap={showMiniMap}
-      readOnly={readOnly}
+      showLineNumbers={args.showLineNumbers}
+      showMiniMap={args.showMiniMap}
```
Comment (hugohaggmark) [inline packages/grafana-ui/src/components/Monaco/CodeEditor.story.internal.tsx:40]: nit: Changing these props in StoryBook does nothing currently. Not sure if it's an issue in the CodeEditor but I would expect something to change when I click the switches. The readonly switch works :)

**Review by hugohaggmark — APPROVED** [review]
Looks good and works as expected without the crash. 🎉

File: packages/grafana-ui/src/components/Monaco/CodeEditor.story.internal.tsx (hunk 1 above)
Comment (thisisobate) [reply packages/grafana-ui/src/components/Monaco/CodeEditor.story.internal.tsx:40]: Yeah...that's weird. Would look into this

**Review by jackw — APPROVED** [review]
Great work on this Uche! 🎉

File: packages/grafana-ui/src/components/Monaco/CodeEditor.story.internal.tsx (hunk 1 above)
Comment (thisisobate) [reply packages/grafana-ui/src/components/Monaco/CodeEditor.story.internal.tsx:40]: Did some investigation but couldn't find anything yet

Comment (thisisobate) [reply packages/grafana-ui/src/components/Monaco/CodeEditor.story.internal.tsx:40]: Oh...just figured.
I think the issue is from the CodeEditor.

Comment (thisisobate) [reply packages/grafana-ui/src/components/Monaco/CodeEditor.story.internal.tsx:40]: Will push a fix now.

Comment (thisisobate) [reply packages/grafana-ui/src/components/Monaco/CodeEditor.story.internal.tsx:40]: For the `showMiniMap` option to work, the value needs to be atleast 100 characters long. So, I'm considering more character to it by default

Comment (hugohaggmark) [reply packages/grafana-ui/src/components/Monaco/CodeEditor.story.internal.tsx:40]: Aaah, good catch 🎉

Comment (hugohaggmark) [reply packages/grafana-ui/src/components/Monaco/CodeEditor.story.internal.tsx:41]: Nevermind this, not super important now.

File: packages/grafana-ui/src/components/Monaco/CodeEditor.tsx [hunk 2]
```diff
@@ -125,6 +125,9 @@ class UnthemedCodeEditor extends React.PureComponent<Props> {
       tabSize: 2,
       codeLens: false,
       contextmenu: false,
+      glyphMargin: true,
+      lineNumbers: 'on',
+      folding: true,
```
Comment (hugohaggmark) [inline packages/grafana-ui/src/components/Monaco/CodeEditor.tsx:130]: I don't think we should change these defaults without checking with lots of people. I think you should revert this change and go ahead and merge this.

**Review by hugohaggmark — APPROVED** [review]
Revert the default options change and then this is good to merge.

File: packages/grafana-ui/src/components/Monaco/CodeEditor.tsx (hunk 2 above)
Comment (thisisobate) [reply packages/grafana-ui/src/components/Monaco/CodeEditor.tsx:130]: alright 👍
